dataactbroker/handlers/accountHandler.py

```diff
@@ -830,45 +886,61 @@ def set_skip_guide(self, session):
+                    "Must include skip_guide parameter",
+                    StatusCode.CLIENT_ERROR
+                )
+            skip_guide = request_dict['skip_guide']
+            if isinstance(skip_guide, bool):    # e.g. from JSON
+                user.skip_guide = skip_guide
+            elif isinstance(skip_guide, str):
+                # param is a string, allow "true" or "false"
+                if skip_guide.lower() == "true":
+                    user.skip_guide = True
+                elif skip_guide.lower() == "false":
+                    user.skip_guide = False
+                else:
+                    raise ResponseException(
+                        "skip_guide must be true or false",
+                        StatusCode.CLIENT_ERROR
+                    )
             else:
-                exc = ResponseException("skip_guide must be true or false", StatusCode.CLIENT_ERROR)
-                return JsonResponse.error(exc, exc.status)
-        else:
-            exc = ResponseException("skip_guide must be a boolean", StatusCode.CLIENT_ERROR)
+                raise ResponseException(
+                    "skip_guide must be a boolean", StatusCode.CLIENT_ERROR)
+        except ResponseException as exc:
             return JsonResponse.error(exc, exc.status)
         sess.commit()
-        return JsonResponse.create(StatusCode.OK,{"message":"skip_guide set successfully","skip_guide":skip_guide})
+        return JsonResponse.create(
+            StatusCode.OK,
+            {"message": "skip_guide set successfully",
+             "skip_guide":skip_guide}
+        )
 
     def email_users(self, system_email, session):
         """ Send email notification to list of users """
         sess = GlobalDB.db().session
-        request_dict = RequestDictionary(self.request)
-        if not (request_dict.exists("users") and request_dict.exists("submission_id") and request_dict.exists("email_template")):
-            exc = ResponseException("Email users route requires users, email_template, and submission_id", StatusCode.CLIENT_ERROR)
+        request_dict = RequestDictionary.derive(self.request)
+        required = ('users', 'submission_id', 'email_template')
+        try:
+            if any(field not in request_dict for field in required):
+                raise ResponseException(
+                    "Email users route requires users, email_template, and "
+                    "submission_id", StatusCode.CLIENT_ERROR
+                )
+        except ResponseException as exc:
             return JsonResponse.error(exc, exc.status)
 
         current_user = sess.query(User).filter(User.user_id == session["name"]).one()
 
-        user_ids = request_dict.getValue("users")
-        submission_id = request_dict.getValue("submission_id")
+        user_ids = request_dict['users']
+        submission_id = request_dict['submission_id']
         # Check if submission id is valid
         self.jobManager.getSubmissionById(submission_id)
```

Comment:
I don't think the logging changes affect that decision. This PR does all of the `try-catch`s to mimic what was happening before (where we were explicitly returning an error `JsonResponse`). Given that `getSubmissionById` is currently raising `ResponseException`, you'll need to decide whether it makes sense to continue to raise that, handle the Sqlalchemy exceptions here (or elsewhere), or use some other strategy. The bit that ties into logging is that `JsonResponse.error` needs to be called within an exception-handing context, if you're planning to use that.